src/status.rs

```diff
@@ -57,11 +60,47 @@ impl StatusUpdater for BuildStatus {
 	}
 }
 
+impl BuildStatusInner {
+	fn set_status(&mut self, worker: usize, status: WorkerStatus) {
+		self.workers[worker] = status;
+		self.dirty = true;
+	}
+
+	fn all_workers_done(&self) -> bool {
```

Comment:
In rust, people usually name 'boolean checks' with `is_`: `is_empty()` instead of just `empty()` etc. Following that, this would be `are_all_workers_done()`.

(I don't care if you change this, I don't always follow that rule either. But just wanted to point this out, as it is different than the convention used by the C++ standard library for example.)